clang/test/dpct/syncthreads.cu

```diff
@@ -149,4 +151,215 @@ __global__ void test9(S4 a) {
   S4 b(a);
   // CHECK:item_ct1.barrier();
   __syncthreads();
-}
\ No newline at end of file
+}
+
+extern __shared__ float extern_local_decl[];
```

Comment:
For device call in global function, we need check all arguments (see if it is global mem access)

Reply:
fixed